File: go/0989-add-to-array-form-of-integer.go
package main

func main() {

}

func addToArrayForm(num []int, k int) []int {
	// Reverse
	for i, j := 0, len(num)-1; i < j; i, j = i+1, j-1 {
		num[i], num[j] = num[j], num[i]
	}
	i := 0

	for k > 0 {
		digit := k % 10

		if i < len(num) {
			num[i] += digit
		} else {
			num = append(num, digit)
		}

		carry := num[i] / 10
		num[i] %= 10

		k /= 10
		k += carry
		i++
	}

	// Reverse
	for i2, j := 0, len(num)-1; i2 < j; i2, j = i2+1, j-1 {
		num[i2], num[j] = num[j], num[i2]
	}

	return num
}

File: javascript/0989-add-to-array-form-of-integer.js
/**
 * @param {number[]} num
 * @param {number} k
 * @return {number[]}
 */
var addToArrayForm = function (num, k) {
    num.reverse();
    let i = 0;

    while (k > 0) {
        const digit = k % 10;
        i < num.length ? (num[i] += digit) : num.push(digit);

        const carry = Math.floor(num[i] / 10);
        num[i] %= 10;

        k = Math.floor(k / 10);
        k += carry;
        i++;
    }

    return num.reverse();
};
